core/tests/functional-tests/test_detection.py

```diff
@@ -0,0 +1,4434 @@
+            "parameters": {"ious": [0.1, 0.6]},
+            "value": 0.0,
+            "type": "AR",
+        },
+        {
+            "label": {"key": "k1", "value": "v1"},
+            "parameters": {"ious": [0.1, 0.6]},
+            "value": 0.5,
+            "type": "AR",
+        },
+        {
+            "parameters": {"label_key": "k1", "ious": [0.1, 0.6]},
+            "value": 0.5,
+            "type": "mAR",
+        },
+        {
+            "parameters": {"label_key": "k2", "ious": [0.1, 0.6]},
+            "value": 0.0,
+            "type": "mAR",
+        },
+    ]
+
+    for m in metrics:
+        if m["type"] not in [
+            "PrecisionRecallCurve",
+            "DetailedPrecisionRecallCurve",
+        ]:
+            assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in metrics
+
+    assert eval_job.ignored_pred_labels == []
+    assert eval_job.missing_pred_labels == []
+
+    result = eval_job
+    result_dict = result.to_dict()
+
+    # duration isn't deterministic, so test meta separately
+    assert result_dict["meta"]["datums"] == 2
+    assert result_dict["meta"]["labels"] == 2
+    assert result_dict["meta"]["annotations"] == 5
+    assert result_dict["meta"]["duration"] <= 5
+    result_dict.pop("meta")
+    result_dict.pop("metrics")
+
+    assert result_dict == {
+        "parameters": {
+            "label_map": {},
+            "metrics_to_return": [
+                enums.MetricType.AP,
+                enums.MetricType.AR,
+                enums.MetricType.mAP,
+                enums.MetricType.APAveragedOverIOUs,
+                enums.MetricType.mAR,
+                enums.MetricType.mAPAveragedOverIOUs,
+            ],
+            "iou_thresholds_to_compute": [0.1, 0.6],
+            "iou_thresholds_to_return": [0.1, 0.6],
+            "recall_score_threshold": 0.0,
+            "pr_curve_iou_threshold": 0.5,
+            "pr_curve_max_examples": 1,
+            "convert_annotations_to_type": None,
+        },
+        "confusion_matrices": [],
+        "ignored_pred_labels": [],
+        "missing_pred_labels": [],
+    }
+
+    #  check that metrics arg works correctly
+    selected_metrics = random.sample(
+        [
+            enums.MetricType.AP,
+            enums.MetricType.AR,
+            enums.MetricType.mAP,
+            enums.MetricType.APAveragedOverIOUs,
+            enums.MetricType.mAR,
+            enums.MetricType.mAPAveragedOverIOUs,
+            enums.MetricType.PrecisionRecallCurve,
+        ],
+        2,
+    )
+    eval_job = evaluate_detection(
+        groundtruths=groundtruths,
+        predictions=predictions,
+        iou_thresholds_to_compute=[0.1, 0.6],
+        iou_thresholds_to_return=[0.1, 0.6],
+        metrics_to_return=selected_metrics,
+    )
+
+    metrics = eval_job.metrics
+    assert set([metric["type"] for metric in eval_job.metrics]) == set(
+        selected_metrics
+    )
+
+
+def test_two_groundtruths_one_datum(
```

Comment:
note that this test shows that you can have multiple GroundTruths per datum without issue. in the API, this operation will throw an error since postgres will try to create the same datum twice (which makes sense). this issue doesn't exist in `valor_core`, so having two GroundTruths with one annotation each is equivalent to having one GroundTruth with two annotations. 

calling this out in case you all want to discuss further. I don't think there's a need to implement another validation check to disallow this behavior, but open to your thoughts